src/vs/workbench/api/common/pluginHostDocuments.ts

```diff
@@ -553,6 +564,9 @@ export class MainThreadDocuments {
 		modelService.onModelRemoved.add(this._onModelRemoved, this, this._toDispose);
 		modelService.onModelModeChanged.add(this._onModelModeChanged, this, this._toDispose);
 
+		this._toDispose.push(eventService.addListener2(FileEventType.FILE_SAVING, (e: LocalFileChangeEvent) => {
+			this._proxy._acceptModelSaving(e.getAfter().resource);
+		}));
```

Comment:
What guarantees to we make when a document is being saved? Can people still modify etc?